src/pygama/raw/orca/orca_SIS3316.py

```
@@ -0,0 +1,229 @@
+            },
+            "crate": {
+                "dtype": "uint8",
+            },
+            "card": {
+                "dtype": "uint8",
+            },
+            "channel": {
+                "dtype": "uint8",
+            },
+            "waveform": {
+                "dtype": "uint16",
+                "datatype": "waveform",
+                "wf_len": 65532,  # max value. override this before initalizing buffers to save RAM
+                "dt": 8,  # override if a different clock rate is use
+                "dt_units": "ns",
+                "t0_units": "ns",
+            },
+        }
+
+        # self.event_header_length = 1 #?
+        self.decoded_values = {}
+        self.skipped_channels = {}
+        super().__init__(
+            header=header, **kwargs
+        )  # also initializes the garbage df (whatever that means...)
+
+    def set_header(self, header: OrcaHeader) -> None:
+        self.header = header
+        import copy
+
+        self.decoded_values = copy.deepcopy(self.decoded_values_template)
+        # print("header", self.header['ObjectInfo']['Crates'])
+        # print("cards", self.header['ObjectInfo']['Crates'][0]['Cards'])
+
+        for card_dict in self.header["ObjectInfo"]["Crates"][0]["Cards"]:
+            if card_dict["Class Name"] == "ORSIS3316Model":
+                card = card_dict["Card"]
+                crate = 0
+                for channel in range(0, 16):
+                    ccc = get_ccc(crate, card, channel)
+                    trace_length = card_dict["rawDataBufferLen"]
+                    self.decoded_values[ccc] = copy.deepcopy(
+                        self.decoded_values_template
+                    )
+
+                    if trace_length <= 0 or trace_length > 2**16:
+                        raise Exception(
+                            "SIS3316ORCADecoder Error: invalid trace_length",
+                            trace_length,
+                        )
```

```suggestion
                        raise RuntimeError("invalid trace_length: ", trace_length)
```
Please raise this if something unexpected happens during decoding at runtime, `Exception` is too generic. `SIS3316ORCADecoder Error: ` is not needed since it will be clear from the stacktrace. Please fix this also in the rest of the code.